jbpm-case-mgmt/jbpm-case-mgmt-impl/src/test/java/org/jbpm/casemgmt/impl/CaseRuntimeDataServiceImplTest.java

```diff
@@ -88,21 +89,12 @@ public void testStartEmptyCaseWithCaseFile() {
             assertEquals(FIRST_CASE_ID, cInstance.getCaseId());
             assertNotNull(cInstance.getCaseFile());
             assertEquals("my first case", cInstance.getCaseFile().getData("name"));
+            
+            List<CaseStatus> statuses = Collections.singletonList(CaseStatus.OPEN);
 
-            Collection<CaseInstance> instances = caseRuntimeDataService.getCaseInstances(new QueryContext());
-            assertNotNull(instances);
-
-            assertEquals(1, instances.size());
-            CaseInstance instance = instances.iterator().next();
-            assertNotNull(instance);
-
-            assertEquals(FIRST_CASE_ID, instance.getCaseId());
-            assertEquals(EMPTY_CASE_P_ID, instance.getCaseDefinitionId());
-            assertEquals("my first case", instance.getCaseDescription());
-            assertEquals(USER, instance.getOwner());
-            assertEquals(ProcessInstance.STATE_ACTIVE, instance.getStatus().intValue());
-            assertEquals(deploymentUnit.getIdentifier(), instance.getDeploymentId());
-            assertNotNull(instance.getStartedAt());
+            assertCaseInstance(caseRuntimeDataService.getCaseInstances(statuses, true, new QueryContext()));
+            assertCaseInstance(caseRuntimeDataService.getCaseInstancesByDefinition(EMPTY_CASE_P_ID, statuses, true, new QueryContext()));
+            assertCaseInstance(caseRuntimeDataService.getCaseInstancesByDeployment(deploymentUnit.getIdentifier(), statuses, true, new QueryContext()));
```

Comment:
Add also a test for getCaseInstancesOwnedBy including withData=true

Reply:
Added in a different place